src/aap_eda/core/utils/k8s_service_name.py

```diff
@@ -0,0 +1,47 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import re
+
+
+class InvalidRFC1035Label(Exception):
+    pass
+
+
+def create_k8s_service_name(activation_name: str) -> str:
+    """Attempt to convert the activation name to a 1035 name.
+
+    Convert dot(.) underscore(_) space( ) to - and convert it to lowercase
+    """
+    name = re.sub(r"[._\s+]", "-", activation_name.lower())
```

Comment:
Can I suggest an improvement in the regex to negate the allowed characters instead?
`r"[^a-z0-9-]", "-"`

Reply:
I feel that replacement is too much. The auto-generated service name will be unreadable if the activation name has many replaced characters.